sharktank/sharktank/layers/configs/llm_configs.py

```diff
@@ -221,6 +272,55 @@ class LlamaModelConfig:
     # A list of layer indices where chunked attention is applied instead of full attention.
     chunked_attention_layers: Optional[set[int]] = None
 
+    # Indices of layers that are MoE.
+    moe_layers: Optional[list[int]] = None
+
+    # Indices of layers that use RoPE after the attention.
+    rope_layers: Optional[list[int]] = None
+
+    use_qk_norm: bool = False
+    # If True, applies normalization to the query and key vectors in attention.
+
+    # In HuggingFace transformers, this field is represented as an int, but it is only ever used as a boolean.
+    # For clarity and correctness, it should be a bool: if True, enables attention temperature tuning.
+    attn_temperature_tuning: Optional[bool] = None
+
+    # Scaling factor applied to attention scores.
+    attn_scale: Optional[float] = None
+
+    # Scaling factor applied as a floor value in attention computations.
+    floor_scale: Optional[int] = None
+
+    # If True, the Feed-Forward Network (FFN) block adds the residual connection at the end.
+    ffn_add_residual: bool = True
+
+    # The size of the model's vocabulary.
+    vocabulary_size: Optional[int] = None
```

Comment:
I was wrong putting this here. It should be in `LlamaHParams` as it is available in GGUF. there also would need to be addition in the `from_gguf_prop` and `to_gguf_props` methods.
The difference is that there it is named `vocab_size`.